### PR TITLE
fetch_sources: give better error for external RPMs

### PR DESCRIPTION
**Pull request by chmeliik**
* OSBS-8362

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by MartinBasti** [comment]
I can document it as I'm working on docs now

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 1]
```diff
@@ -168,6 +168,10 @@ def get_srpm_urls(self, sigkeys=None, insecure=False):
         for rpm_id, rpm_build_id in rpm_build_ids.items():
             self.log.debug('Resolving SRPM for RPM ID: %s', rpm_id)
             rpm_hdr = self.session.getRPMHeaders(rpm_id, headers=['SOURCERPM'])
+            if 'SOURCERPM' not in rpm_hdr:
+                msg = ('RPM comes from an external repo (RPM ID: {}). '
```
Comment (athos-ribeiro) [inline atomic_reactor/plugins/pre_fetch_sources.py]: Is it possible that an RPM will lack the sourcerpm header for a different reason?
The entries from getRPMs contain a `external_repo_name` that do describe if the repository is `INTERNAL` or sth else.

Comment (chmeliik) [reply atomic_reactor/plugins/pre_fetch_sources.py]: That seems more solid, thanks

**Comment by chmeliik** [comment]
Now checking `external_repo_name` key for rpms, I left the check for missing 'SOURCERPM' header in anyway, not sure if it is still wanted

**Comment by athos-ribeiro** [comment]
> Now checking `external_repo_name` key for rpms, I left the check for missing 'SOURCERPM' header in anyway, not sure if it is still wanted

I am not aware of other reasons for the SOURCERPM header to be missing. (Maybe if the RPM is imported through some other means). I see no reasons for not keeping the check though.

**Comment by MartinBasti** [comment]
Release notes:
```
- source containers: check that only internal RPMs are used
```

**Pull request merged**